File: client/src/lib/services/API/player/post.ts
import { type Player } from "$lib/types/Player";

export default async (name: string) : Promise<Player> => {
  const response = await fetch('/api/v1/player', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
        name
    })
  });
  
  if (!response.ok) throw Error("Failed to create player.");

  const { player } = await response.json();

  return player;
}

File: api/router/endpoints/v1/player/Post.go
package player

import (
	"api/database"
	gameHelpers "api/lib/game"
	"net/http"

	"github.com/gin-gonic/gin"
)

type PostPlayerInput struct {
	Name string `json:"name"`
}

func Post(context *gin.Context) {
	var input PostPlayerInput
	validationError := context.ShouldBindJSON(&input)
	if validationError != nil {
		context.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
		return
	}

	// Create a player.
	database := database.Get()
	player, playerError := gameHelpers.CreatePlayer(input.Name, database)
	if playerError != nil {
		context.JSON(http.StatusInternalServerError, gin.H{"error": "Database error"})
	}

	// Set the player cookie.
	gameHelpers.SetPlayerCookie(context, *player)

	context.JSON(http.StatusOK, gin.H{"player": *player})
}
